wasm/execution/configuration.py

```diff
@@ -218,8 +214,11 @@ def __init__(self, store: Store) -> None:
     def execute(self) -> Tuple[TValue, ...]:
         from wasm.logic import OPCODE_TO_LOGIC_FN
 
-        while self.has_active_frame:
-            instruction = next(self.instructions)
+        while True:
+            try:
+                instruction = next(self.instructions)
+            except AttributeError:
```

Comment:
Seems a little weird in this context to catch an `AttributeError`. It requires knowing the internals of how the iterator works. Maybe catch and re-raise a different exception inside the iterator? We could potentially even just raise a `StopIteration`, then this whole loop reduces to: `instruction = last(self.instructions)`

Reply:
I've updated a number of things, but kept this section the same for the following reasons.

The `AttributeError` now actually comes from access to `self._instructions`.  I've moved the `next` (now `__next__()`) call down outside of the `try/except` to make it clear where the exception comes from.

The reason for this is that I've now extended the memoization to the `Configuration` object itself to reap some further performance gains.  Part of this was changing from use of an `@property` method to a direct property to reduce the overhead of the additional call frame caused by computed properties.

I've added a large comment in the body of this function outlining the parts that are less readable or *pythonic* in favor of performance to make it clearer why it is written the way that it is.